File: solutions/06graph/36secho.py
from typing import List
from itertools import combinations
class Solution:
    def combinationSum(self, candidates: List[int], target: int) -> List[List[int]]:
      combis = []
      result = []
      for i in range(1, len(candidates)+1):
        combis.append(list(map(list, combinations(candidates, i))))
      for i in range(len(combis)):
        for j in range(len(combis[i])):
          if ((target % sum(combis[i][j])) == 0):
            print(combis[i][j])
    combinationSum("",[2,3,6,7],7)

File: solutions/06graph/36secho.js
/**
 * @param {number[]} candidates
 * @param {number} target
 * @return {number[][]}
 */
var combinationSum = function (candidates, target) {
  let result = [];
  const dfs = (element, callSize) => {
    let total = element.reduce((sum, curr) => sum + curr, 0);
    if (total > target) {
      element.pop();
      return;
    }
    if (total == target) {
      result.push(element);
      return;
    }
    for (let i of candidates) {
      element.push(i);
      dfs(element, callSize + 1);
    }
  };
  for (let i of candidates) {
    dfs([i], 0);
  }
  return result;
};

console.log(combinationSum([2, 3, 5], 8));
